packages/fuels-contract/src/contract.rs

```diff
@@ -22,6 +23,12 @@ use fuels_signers::provider::Provider;
 use fuels_signers::{LocalWallet, Signer};
 use std::marker::PhantomData;
 
+#[async_trait]
+pub trait Deployable {
```

Comment:
I don't understand the point of this trait if it is used only once. Traits are used, afaik, to have a sort of generic interface. In our case, only `Contracts` are `Deployable` basically, so it does not make sense IMO to create a whole trait for it, especially if you have to use a new crate for the trait to exist.

Reply:
I've added some more context as to why this is needed in a comment. Please take a look and let me know what you think.

Reply:
Actually you were right, we can do without the trait here. I went trough quite a few iterations and the trait is just a remnant. There is still one advantage to it, if the user defines functions `deploy_compiled()`, `new()`, `get_binary_filepath()` on the sway side, abigen will fail. This way they could coexist and be accessed via fully qualified syntax.

Reply:
`new`, other binding methods and any sway methods are generated within the same `impl` block. I ran a quick test with `new` defined in the json ABI to confirm we get a collision. I would therefore move new to the trait impl and keep it for now, maybe with a different name. If you know of a cleaner way to handle this, let me know.